content/kubernetes-platform-services/artifact-storage/chartmuseum.md

```diff
@@ -0,0 +1,128 @@
+
+## Provision Chamber Resources
+
+Change directory to `/conf/kops-aws-platform` and run these commands to provision the `chart-repo` backend.
+```bash
+init-terraform
+terraform plan
+terraform apply
+```
+
+From the Terraform outputs, copy the following values into the environment variables
+* `kops_chart_repo_bucket_bucket_id` -> `CHARTMUSEUM_STORAGE_AMAZON_BUCKET`
+* `kops_chart_repo_bucket_role_name` -> `CHARTMUSEUM_IAM_ROLE`
+
+{{% include-code-block title="terraform apply" file="kubernetes-platform-services/artifact-storage/examples/terraform-apply-chart-repo.txt" %}}
+
+In the example the bucket name is `kops_chart_repo_bucket_bucket_id = example-staging-chart-repo`.
+IAM role is `kops_chart_repo_bucket_role_name = example-staging-chart-repo`.
+
+## Install Chart
+
+To install the `chartmuseum`, you will need to define the `hostname`, which is the FQHN used to access the `chartmuseum`.
+
+In our example, we use `charts.us-west-2.staging.example.com` as the FQHN. Replace this with an appropriate value to suit your specific project.
+
+You can install `chartmuseum` in a few different ways, but we recommend using the [Master Helmfile](https://github.com/cloudposse/geodesic/blob/master/rootfs/conf/kops/helmfile.yaml).
+
+### Install with Master Helmfile
+
+Master Helmfile provides two releases of chartmuseum:
+* `charts` - Chartmuseum that serve charts
+* `charts-api` - Chartmuseum that provide api gateway to publish charts.
+
+These releases share the same environment variables.
+`charts-api` gateway will be available on the subdomain `api` for the FQHN.
+In our example it would be `api.charts.us-west-2.staging.example.com`.
+
+To install releases follow these instructions:
+1. Set the `CHARTMUSEUM_STORAGE_AMAZON_BUCKET` secret with chamber to the value copied from the Terraform output
+2. Set the `CHARTMUSEUM_STORAGE_AMAZON_REGION` secret with chamber
+3. Set the `CHARTMUSEUM_IAM_ROLE` secret with chamber to the value copied from the Terraform output
+4. Set the `CHARTMUSEUM_INGRESS` secret with chamber provided by [Nginx ingress]({{< relref "kubernetes-backing-services/ingress/nginx-ingress-controller.md" >}})
+5. Set the `CHARTMUSEUM_HOSTNAME` secret with chamber
+6. Run the following commands to install `chartmuseum`.
+
+{{% dialog type="code-block" icon="fa fa-code" title="Install chartmuseum" %}}
+```
+chamber write kops CHARTMUSEUM_STORAGE_AMAZON_BUCKET example-staging-chart-repo
+chamber write kops CHARTMUSEUM_STORAGE_AMAZON_REGION us-west-2
+chamber write kops CHARTMUSEUM_IAM_ROLE example-staging-chart-repo
+chamber write kops CHARTMUSEUM_INGRESS ingress.us-west-2.staging.example.com
+chamber write kops CHARTMUSEUM_HOSTNAME charts.us-west-2.staging.example.com
+chamber exec kops -- helmfile -f /conf/kops/helmfile.yaml --selector namespace=kube-system,chart=chartmuseum sync
+chamber exec kops -- helmfile -f /conf/kops/helmfile.yaml --selector namespace=kube-system,chart=chartmuseum-api sync
```

Comment:
Do we need to doc both here? I used `chamber exec kops -- helmfile -f helmfile.yaml --selector chart=chartmuseum sync` and it installs both chartmuseum and chartmusem-api into kube-system without issue. Including both for the sake of clarity makes sense, just curious if I missed something important.

Reply:
Yea, I agree with @MoonMoon1919. Actually, I think this was something we fixed in the `helmfile.yaml`, but didn't update yet in the docs.

Reply:
https://github.com/cloudposse/geodesic/pull/162/files#diff-02bcde86e152646f0e55c8d0667bd3f0R156